cmd/lotus-seal-worker/main.go

```diff
@@ -317,3 +330,42 @@ var runCmd = &cli.Command{
 		return srv.Serve(nl)
 	},
 }
+
+func watchMinerConn(ctx context.Context, cctx *cli.Context, nodeApi api.StorageMiner) {
+	go func() {
+		closing, err := nodeApi.Closing(ctx)
+		if err != nil {
+			log.Errorf("failed to get remote closing channel: %+v", err)
+		}
+
+		select {
+		case <-closing:
+		case <-ctx.Done():
+		}
+
+		if ctx.Err() != nil {
```

Comment:
shouldnt this check be inside the context case in the select above?

Reply:
No, because the `closing` channel can also be closed by the context, which could be racy - context on the other hand can only by closed with C-c